### PR TITLE
Find maximum value

### PR DESCRIPTION
**Pull request by paronicholas**
- [x] Top-level README “Table of Contents” is updated
- [x] Feature tasks for this challenge are completed
- [x] Unit tests written and passing
    - [x] “Happy Path” - Expected outcome
    - [x] Expected failure
    - [x] Edge Case (if applicable/obvious)
- [x] README for this challenge is complete
    - [x] Summary, Description, Approach & Efficiency, Solution, Resources
    - [x] Link to code
    - [x] Link to tests

### CODE REVIEW

File: code401challenges/src/main/java/tree/Tree.java [hunk 1]
```diff
@@ -92,4 +92,24 @@ public boolean isEmpty() {
...
+    return nodeVal instanceof Integer;
+  }
+  
+  public int findMaximumValue() {
+    if (this.isEmpty()) {
+      throw new NoSuchElementException();
+    } else if (!isTypeInteger(this.getRoot().getValue())) {
+      throw new IllegalStateException("The Tree must be cast to the type of 'Integer' to use this method");
+    }
+    List<T> traversedTree = this.inOrder();
```
Comment (evanslaton) [inline code401challenges/src/main/java/tree/Tree.java]: Nice!
